src/shared_modules/indexer_connector/src/monitoring.hpp

```diff
@@ -148,13 +151,28 @@ class Monitoring final
                                         {
                                             value.second = false;
                                         }
+                                        logDebug2(IC_NAME,
+                                                  "Cluster health status '%s' for server '%s'",
+                                                  Utils::toUpperCase(fields.at(HealthCheckColumns::STATUS)).c_str(),
+                                                  value.first.c_str());
                                     }
                                     else
                                     {
+                                        logDebug2(IC_NAME,
+                                                  "Invalid response from /_cat/health?v: %s",
+                                                  originalResponse.c_str());
                                         value.second = false; // LCOV_EXCL_LINE
                                     }
                                 },
-                                [&](const std::string& error, const long /*statusCode*/) { value.second = false; },
+                                [&](const std::string& error, const long statusCode)
+                                {
+                                    std::string errorMessage = error;
+
+                                    errorMessage += " (Status code: " + std::to_string(statusCode) + ")";
```

Comment:
Improve (design): This could be a const
```suggestion
                                    const std::string errorMessage = error + ". Status code: " + std::to_string(statusCode);
```